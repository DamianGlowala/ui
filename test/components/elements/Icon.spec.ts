import { describe, it, expect } from 'vitest'
import { UIcon } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Icon', () => {
  it.each([
    ['basic case', {}],
    // Cas où dynamic est à false
    ['with name and dynamic', { props: { name: 'i-heroicons-academic-cap', dynamic: true } }],
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof Icon.props>) => {
    if (options !== undefined) {
      options.slots = options.slots || { default: () => 'label' }
      options.slots.default = options.slots.default || (() => 'label')
    }
    const html = await ComponentRender(nameOrHtml, options, UIcon)
    expect(html).toMatchSnapshot()
  })
})
