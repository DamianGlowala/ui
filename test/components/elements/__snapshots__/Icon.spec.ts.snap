// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Icon > renders basic case correctly 1`] = `"<span class=""></span>"`;

exports[`Icon > renders with name correctly 1`] = `"<span class="i-heroicons-academic-cap"></span>"`;
